### PR TITLE
[ADD] CRM: merge similar leads backport

### PR DESCRIPTION
**Pull request by pabr-odoo**
Task card: https://www.odoo.com/odoo/project/3835/tasks/4044525

Backporting the 17.0 doc "Merge similar leads and opportunities" to 16.0.

### CODE REVIEW

**Comment by pabr-odoo** [comment]
Hey Zac, tagging you for first review of this backport from 17.0 to 16.0. Thanks!

**Comment by pabr-odoo** [comment]
Hey @jero-odoo, tagging you for first review of this backport from 17.0 to 16.0 whenever you get the chance. Thanks!

**Review by jero-odoo — APPROVED** [review]
Hey @pabr-odoo this looks good to go to the next round (I confirmed the info from 17 works for 16 so we should be all good) :+1: Approving for final review!

**Comment by pabr-odoo** [comment]
Hey @ksc-odoo just passing this along for final review whenever you get the chance. Thanks!

File: content/applications/sales/crm/pipeline/merge_similar.rst [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+Comparing similar leads and opportunities
+-----------------------------------------
+
+To compare the details of similar leads/opportunities, navigate to :menuselection:`CRM app -->
+Pipeline` or :menuselection:`CRM app --> Leads`. Open a lead or opportunity, and click the
+:guilabel:`Similar Leads` smart button. Doing so opens a Kanban view that only displays similar
+leads/opportunities. Click on a card to view the details for the lead/opportunity, and confirm if
+they should be merged.
+
+Merging similar leads and opportunities
```
Comment (ksc-odoo) [inline content/applications/sales/crm/pipeline/merge_similar.rst:34]:
```suggestion
Merge similar leads and opportunities
```

File: content/applications/sales/crm/pipeline/merge_similar.rst [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+.. image:: merge_similar/similar-smart-button.png
+   :align: center
+   :alt: An opportunity record with emphasis on the Similar Leads smart button.
+
+Comparing similar leads and opportunities
+-----------------------------------------
+
+To compare the details of similar leads/opportunities, navigate to :menuselection:`CRM app -->
+Pipeline` or :menuselection:`CRM app --> Leads`. Open a lead or opportunity, and click the
+:guilabel:`Similar Leads` smart button. Doing so opens a Kanban view that only displays similar
```
Comment (ksc-odoo) [inline content/applications/sales/crm/pipeline/merge_similar.rst:30]:
```suggestion
:guilabel:`Similar Leads` smart button. Doing so opens an :guilabel:`Opportunities' page, in Kanban view, that only displays similar
```

File: content/applications/sales/crm/pipeline/merge_similar.rst [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+Merging similar leads and opportunities
+=======================================
+
+.. important::
+   When merging, Odoo gives priority to whichever lead/opportunity was created in the system first,
+   merging the information into the first created lead/opportunity. However, if a lead and an
+   opportunity are being merged, the resulting record is referred to as an opportunity, regardless
+   of which record was created first.
+
+After confirming that the leads/opportunities should be merged, return to the Kanban view using the
```
Comment (ksc-odoo) [inline content/applications/sales/crm/pipeline/merge_similar.rst:43]:
```suggestion
After confirming the leads/opportunities should be merged, return to the :guilabel:`Opportunities' page, in Kanban view, using the
```

File: content/applications/sales/crm/pipeline/merge_similar.rst [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+breadcrumb link, or by clicking the :guilabel:`Similar Leads` smart button. Click the
```
Comment (ksc-odoo) [inline content/applications/sales/crm/pipeline/merge_similar.rst:44]:
```suggestion
breadcrumb link, or by clicking the :guilabel:`Similar Leads` smart button on the opportunity form. Click the
```

File: content/applications/sales/crm/pipeline/merge_similar.rst [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+:icon:`oi-view-list` :guilabel:`(list)` icon to change to list view.
+
+Check the box on the left of the page for the leads/opportunities to be merged. Then, click the
+:icon:`fa-cog` :guilabel:`Action` icon at the top of the page, to reveal a drop-down menu. From
+that drop-down menu, select the :guilabel:`Merge` option to merge the selected opportunities or
+leads.
```
Comment (ksc-odoo) [inline content/applications/sales/crm/pipeline/merge_similar.rst:50]:
```suggestion
Tick the checkbox on the left of the page for the leads/opportunities to be merged. Then, click the
:icon:`fa-cog` :guilabel:`Action` icon at the top of the page, to reveal a drop-down menu. From
that drop-down menu, select the :guilabel:`Merge` option to merge the selected opportunities or
leads.
```

File: content/applications/sales/crm/pipeline/merge_similar.rst [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+When :guilabel:`Merge` is selected from the :icon:`fa-cog` :guilabel:`Action` drop-down menu, a
+:guilabel:`Merge` pop-up modal appears. In that pop-up modal, under the :guilabel:`Assign
+opportunities to` heading, optionally select a :guilabel:`Salesperson` and :guilabel:`Sales Team`
```
Comment (ksc-odoo) [inline content/applications/sales/crm/pipeline/merge_similar.rst:54]:
```suggestion
opportunities to` section, optionally select a :guilabel:`Salesperson` and :guilabel:`Sales Team`
```

File: content/applications/sales/crm/pipeline/merge_similar.rst [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+from the appropriate drop-down menus.
+
+Below those fields, the leads/opportunities to merge are listed, along with their related
+information. To add more, click the :guilabel:`Add a line button`, check the box next to the desired
```
Comment (ksc-odoo) [inline content/applications/sales/crm/pipeline/merge_similar.rst:58]:
```suggestion
information. To add more, click :guilabel:`Add a line`, tick the checkbox next to the desired
```

File: content/applications/sales/crm/pipeline/merge_similar.rst [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+leads/opportunities, then click the :guilabel:`Select` button. The additional selections now appear
```
Comment (ksc-odoo) [inline content/applications/sales/crm/pipeline/merge_similar.rst:59]:
```suggestion
leads/opportunities from the resulting :guilabel:`Add: Leads/Opportunities' pop-up modal, then click the :guilabel:`Select` button. The additional selections now appear
```

File: content/applications/sales/crm/pipeline/merge_similar.rst [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+in the list. To merge those selected leads/opportunities, click :guilabel:`Merge`.
+
+Once all leads/opportunities are selected, click the :guilabel:`Merge` button. The pop-up window
```
Comment (ksc-odoo) [inline content/applications/sales/crm/pipeline/merge_similar.rst:62]:
```suggestion
The pop-up window
```
I believe you just mentioned the 'Merge' button step in line 60

File: content/applications/sales/crm/pipeline/merge_similar.rst [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+There may be instances where a similar lead or opportunity is identified, but should *not* be
+merged. These circumstances vary, based on the processes of the sales team and organization. Some
+potential scenarios are listed below.
+
+Lost leads
+----------
+
+If a lead/opportunity has been marked as :doc:`lost <lost_opportunities>`, it can still be merged
+with an active lead or opportunity. The resulting lead/opportunity is marked active, and added to
+the pipeline.
```
Comment (ksc-odoo) [inline content/applications/sales/crm/pipeline/merge_similar.rst:85]: Perhaps a quick concluding sentence explaining WHY this scenario shouldn't result in a merge could be beneficial to the reader.

File: content/applications/sales/crm/pipeline/merge_similar.rst [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+
+Different contact within an organization
+----------------------------------------
+
+Leads/opportunities from the same organization, but with different points of contact, may not have
+the same needs. In this case, it is beneficial to *not* merge these records, though assigning the
+same salesperson, or sales team, can prevent duplicated work and miscommunication.
+
+Existing duplicates with more than one salesperson
```
Comment (ksc-odoo) [inline content/applications/sales/crm/pipeline/merge_similar.rst:94]:
```suggestion
Duplicates with more than one salesperson
```

File: content/applications/sales/crm/pipeline/merge_similar.rst [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+--------------------------------------------------
+
+If more than one lead/opportunity exists in the database, there may be multiple salespeople assigned
+to them, who are actively working on them independently. While these leads/opportunities may need
+to be managed separately, it is recommended that any affected salespeople be tagged in an internal
+note for visibility.
+
+Contact information is similar but not exact
```
Comment (ksc-odoo) [inline content/applications/sales/crm/pipeline/merge_similar.rst:102]:
```suggestion
Contact information is similar, but not identical
```

File: content/applications/sales/crm/pipeline/merge_similar.rst [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+--------------------------------------------
+
+Similar leads and opportunities are identified by comparing the email addresses and phone numbers of
+the associated contacts. However, if the email address is *similar*, but not *exact*, they may need
```
Comment (ksc-odoo) [inline content/applications/sales/crm/pipeline/merge_similar.rst:106]:
```suggestion
the associated contacts. However, if the email address is similar, but not identical, they may need
```

File: content/applications/sales/crm/pipeline/merge_similar.rst [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+to remain independent.
+
+.. example::
+   Three different leads were added to the pipeline and assigned to different salespeople. They
```
Comment (ksc-odoo) [inline content/applications/sales/crm/pipeline/merge_similar.rst:110]:
```suggestion
   Three different leads were added to the pipeline, and assigned to different salespeople. They
```

File: content/applications/sales/crm/pipeline/merge_similar.rst [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+   were identified as *Similar Leads* due to the email addresses of the contacts.
```
Comment (ksc-odoo) [inline content/applications/sales/crm/pipeline/merge_similar.rst:111]:
```suggestion
   were identified as *Similar Leads*, due to the email addresses of the contacts.
```

**Review by ksc-odoo — APPROVED** [review]
Hey @pabr-odoo -- just finished my Final Review. Nice job. There are a handful of comments/feedback/etc. that requires your attention. Once you implement the necessary adjustments, feel free to tag this for Tech Review. Thanks! 👍

**Comment by pabr-odoo** [comment]
Hey @samueljlieber just tagging you on this for tech review whenever you can get to it. Thanks so much!

**Review by samueljlieber — APPROVED** [review]
Looks good to me @pabr-odoo, nice job with this backport! Thank you!
.....
@robodoo up to saas-16.4 r+

**Comment by samueljlieber** [comment]
@robodoo fw=no r+

**Pull request closed without merging**